workbench/public/components/Main/main.tsx

```diff
@@ -203,6 +207,9 @@ export class Main extends React.Component<MainProps, MainState> {
     };
 
     this.httpClient = this.props.httpClient;
+    this.updateSQLQueries = _.debounce(this.updateSQLQueries, 250).bind(this);
```

Comment:
Would there be a delay if you debounce state update when editor updates?

Reply:
Yes there would be a minor delay of 250 millisecond from every call of this function, I think it's necessary and 250 ms is acceptable.